keymap_drawer/parse/rmk.py

```diff
@@ -0,0 +1,408 @@
+                "Enter": "RET", "Space": "SPC", "__": "",
+            }
+            
+            display_key = rmk_keycode_map.get(key, key)
+            mapped = LayoutKey.from_key_spec(
+                self.cfg.zmk_keycode_map.get(display_key, display_key)
+            )
+            
+            if no_shifted:
+                mapped.shifted = ""
+            if mods:
+                mapped.apply_formatter(lambda k: self.format_modified_keys(k, mods))
+            return mapped
+
+        # Handle empty/transparent keys
+        if not binding or binding == "__":
+            return LayoutKey()
+
+        # Parse the binding
+        if "(" in binding and binding.endswith(")"):
+            # Handle function-style bindings like MT(C, LCtrl) or LT(2, Backspace)
+            func_match = re.match(r"(\w+)\((.+)\)", binding)
+            if func_match:
+                func_name, params = func_match.groups()
+                param_list = [p.strip() for p in params.split(",")]
+                
+                if func_name == "MT" and len(param_list) == 2:
+                    # Mod-tap: MT(tap_key, hold_modifier)
+                    tap_key = mapped(param_list[0])
+                    hold_mod = mapped(param_list[1])
+                    return LayoutKey(tap=tap_key.tap, hold=hold_mod.tap, shifted=tap_key.shifted)
+                
+                elif func_name == "LT" and len(param_list) == 2:
+                    # Layer-tap: LT(layer_number, tap_key)
+                    try:
+                        layer_num = int(param_list[0])
+                        tap_key = mapped(param_list[1])
+                        
+                        if current_layer is not None:
+                            self.update_layer_activated_from([current_layer], layer_num, key_positions)
+                        
+                        return LayoutKey(
+                            tap=tap_key.tap,
+                            hold=self.layer_legends[layer_num] if layer_num < len(self.layer_legends) else f"L{layer_num}",
+                            shifted=tap_key.shifted
+                        )
+                    except (ValueError, IndexError):
+                        pass
+                
+                elif func_name in ("OSM", "OSL"):
+                    # One-shot modifier or layer
+                    if len(param_list) == 1:
+                        target = mapped(param_list[0])
+                        sticky_label = self.cfg.sticky_label if func_name == "OSM" else target.tap
+                        return LayoutKey(tap=target.tap, hold=sticky_label)
+                
+                elif func_name == "SHIFTED" and len(param_list) == 1:
+                    # Shifted version of a key
+                    base_key = mapped(param_list[0])
+                    return LayoutKey(tap=f"S({base_key.tap})")
+                
+                elif func_name == "WM" and len(param_list) == 2:
```

Comment:
Just checking, are these generic macros supported by RMK or specific to your keymap?